### PR TITLE
[embeddedbroker] bump moquette and fix corrupt database

### LINKED ISSUE #5527: [mqttembeddedbroker] mqttembedded.bin file is locked
Dear @davidgraeff!
The latest Snapshot builds (at least S1578-1582) cause an issue with the embedded MQTT broker. I would be glad if you could have a look.

What I have tried so far: 

- Updated to the most recent JDK zulu8.36.0.152-ca-jdk1.8.0_202
- Uninstalled and reinstalled the MQTT Embedded Broker (MQTT Broker Moquette) and MQTT Binding via PaperUI.

Best regards,
Jan

Shortened log below:
```
2019-04-29 19:42:36.873 [ERROR] [org.openhab.io.mqttembeddedbroker   ] - bundle org.openhab.io.mqttembeddedbroker:2.5.0.201904290415 (228)[org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService(209)] : The activate method has thrown an exception
java.lang.IllegalArgumentException: Could not deserialize [-84, -19, 0, 5, 115, 114, 0, 38, 105, 111, 46, 109, 111, 113, 117, 101, 116, 116, 101, 46, 98, 114, 111, 107, 101, 114, 46, 115, 117, 98, 115, 99, 114, 105, 112, 116, 105, 111, 110, 115, 46, 84, 111, 112, 105, 99, 0, 8, -86, 19, 52, -4, -125, -66, 2, 0, 1, 76, 0, 5, 116, 111, 112, 105, 99, 116, 0, 18, 76, 106, 97, 118, 97, 47, 108, 97, 110, 103, 47, 83, 116, 114, 105, 110, 103, 59, 120, 112, 116, 0, 18, 104, 111, 109, 105, 101, 47, 49, 119, 105, 114, 101, 47, 36, 110, 111, 100, 101, 115] [1.4.199/0]
	at org.h2.mvstore.DataUtils.newIllegalArgumentException(DataUtils.java:857) ~[?:?]
	at org.h2.mvstore.type.ObjectDataType.deserialize(ObjectDataType.java:376) ~[?:?]
	at org.h2.mvstore.type.ObjectDataType$SerializedObjectType.read(ObjectDataType.java:1547) ~[?:?]
.
.
.
	at org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340) [?:?]

Caused by: java.lang.ClassNotFoundException: io.moquette.broker.subscriptions.Topic cannot be found by com.h2database.mvstore_1.4.199

	at org.eclipse.osgi.internal.loader.BundleLoader.findClassInternal(BundleLoader.java:484) ~[?:?]
.
.
.
	at org.h2.mvstore.type.ObjectDataType.deserialize(ObjectDataType.java:374) ~[?:?]

	... 91 more

2019-04-29 19:42:37.975 [INFO ] [roker.internal.EmbeddedBrokerService] - Broker persistence file: mqttembedded.bin

2019-04-29 19:42:37.719 [WARN ] [org.openhab.io.mqttembeddedbroker   ] - FrameworkEvent WARNING - org.openhab.io.mqttembeddedbroker

org.osgi.framework.ServiceException: org.apache.felix.scr.impl.manager.SingleComponentManager.getService() returned a null service object

	at org.eclipse.osgi.internal.serviceregistry.ServiceFactoryUse.factoryGetService(ServiceFactoryUse.java:232) ~[?:?]
.
.
.
org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340) [?:?]
2019-04-29 19:42:38.410 [ERROR] [org.openhab.io.mqttembeddedbroker   ] - bundle org.openhab.io.mqttembeddedbroker:2.5.0.201904290415 (228)[org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService(209)] : The activate method has thrown an exception
java.lang.IllegalStateException: The file is locked: nio:/var/lib/openhab2/mqttembedded.bin [1.4.199/7]
	at org.h2.mvstore.DataUtils.newIllegalStateException(DataUtils.java:883) ~[?:?]
	at org.h2.mvstore.FileStore.open(FileStore.java:166) ~[?:?]
	at org.h2.mvstore.MVStore.<init>(MVStore.java:390) ~[?:?]
	at org.h2.mvstore.MVStore$Builder.open(MVStore.java:3343) ~[?:?]
	at io.moquette.persistence.H2Builder.initStore(H2Builder.java:40) ~[?:?]
	at io.moquette.broker.Server.startServer(Server.java:166) ~[?:?]
	at org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService.startEmbeddedServer(EmbeddedBrokerService.java:307) ~[?:?]
	at org.openhab.io.mqttembeddedbroker.internal.EmbeddedBrokerService.initialize(EmbeddedBrokerService.java:192) ~[?:?]
.
.
.
org.eclipse.osgi.framework.eventmgr.EventManager$EventThread.run(EventManager.java:340) [?:?]
Caused by: java.nio.channels.OverlappingFileLockException
	at sun.nio.ch.SharedFileLockTable.checkList(FileLockTable.java:255) ~[?:?]
	at sun.nio.ch.SharedFileLockTable.add(FileLockTable.java:152) ~[?:?]
	at sun.nio.ch.FileChannelImpl.tryLock(FileChannelImpl.java:1107) ~[?:?]
	at org.h2.store.fs.FileNio.tryLock(FilePathNio.java:121) ~[?:?]
	at java.nio.channels.FileChannel.tryLock(FileChannel.java:1155) ~[?:?]
	at org.h2.mvstore.FileStore.open(FileStore.java:163) ~[?:?]
	... 52 more

```

### PR DESCRIPTION
**Pull request by J-N-K**
Fixes #5527

See https://community.openhab.org/t/solved-mqtt-embedded-crashed-will-not-restart/83342 and https://community.openhab.org/t/solved-mqtt-broker-moquette-offline-after-upgrade-to-2-5m3/83074

### CODE REVIEW

File: bundles/org.openhab.io.mqttembeddedbroker/src/main/java/org/openhab/io/mqttembeddedbroker/internal/EmbeddedBrokerService.java [hunk 1]
```diff
@@ -303,13 +304,21 @@ public void startEmbeddedServer(@Nullable Integer portParam, boolean secure, @Nu
...
+                        authorizer);
+            } else {
+                server.startServer(new MemoryConfig(properties), null, null, authentificator, authorizer);
+            }
+        } catch (IllegalArgumentException e) {
+            if (e.getMessage().contains("Could not deserialize")) {
+                Path persistenceFilePath = Paths.get((new File(persistenceFilename)).getAbsolutePath());
+                logger.warn("persistence corrupt: {}, deleting {}", e.getMessage(), persistenceFilePath);
+                Files.delete(persistenceFilePath);
+                throw  new IllegalStateException("Could not start broker (persistence corrupted)");
```
Comment (davidgraeff) [inline bundles/org.openhab.io.mqttembeddedbroker/src/main/java/org/openhab/io/mqttembeddedbroker/internal/EmbeddedBrokerService.java]: Nice thanks for this patch.
Should it be retried to start the server after the persistence file has been removed?

**Comment by J-N-K** [comment]
Resolution fails due to caching issues. Works fine locally.

**Comment by davidgraeff** [comment]
I have erased travis cache (50 GB!!!!! 1.2 GB per PR and per branch. The cache should be the same on all PRs. Our current cache solution clearly does not work as intended!).
And also triggered a rebuild.

**Comment by J-N-K** [comment]
There is another bug report in moquette which I would like to check before merging this.

**Comment by J-N-K** [comment]
Travis fails unrelated in HueEmulation test

**Comment by davidgraeff** [comment]
Yes looks like it. All mqtt related tests are passing.

**Pull request merged**